scripts/split_territory.rb

```diff
@@ -0,0 +1,175 @@
+
+      agent_in_new_territory = territory_ids_from_agent_organisations.include?(new_territory.id)
+      agent_in_old_territory = territory_ids_from_agent_organisations.include?(old_territory.id)
+
+      if agent_in_new_territory
+        if agent_in_old_territory
+          puts "Création de nouveaux AgentTerritorialAccessRights pour #{agent_territorial_access_right.agent.email}"
+          new_access_right = agent_territorial_access_right.dup
+          new_access_right.update!(territory_id: new_territory.id)
+        else
+          puts "Changement de territoire pour #{agent_territorial_access_right.agent.email}"
+          agent_territorial_access_right.update!(territory: new_territory)
+        end
+      end
+    end
+
+    puts "#{AgentTerritorialAccessRight.where(territory: new_territory).count} agents dans le nouveau territoire"
+  end
+
+  def move_or_duplicate_agent_territorial_roles
+    puts "\n\n## Déplacement des Admins de territoires"
+    AgentTerritorialRole.where(territory: old_territory).each do |agent_territorial_role|
+      agent = agent_territorial_role.agent
+      territory_ids_from_agent_organisations = agent.organisations.pluck(:territory_id)
+
+      agent_in_new_territory = territory_ids_from_agent_organisations.include?(new_territory.id)
+      agent_in_old_territory = territory_ids_from_agent_organisations.include?(old_territory.id)
+
+      if agent_in_new_territory
+        if agent_in_old_territory
+          puts "Création de nouveaux AgentTerritorialRole pour #{agent.email}"
+          AgentTerritorialRole.create!(territory: new_territory, agent: agent)
+        else
+          puts "Changement de territoire pour l'admin de territoire #{agent.email}"
+          agent_territorial_role.update!(territory: new_territory)
+        end
+      end
+    end
+  end
+
+  def move_or_duplicate_motif_categories_territories
+    puts "\n\n## Déplacement ou création des catégories de motifs\n"
+    MotifCategoriesTerritory.where(territory: old_territory).each do |motif_categories_territory|
+      present_in_old_territory = motif_category_present_in_territory?(motif_categories_territory.motif_category_id, old_territory)
+      present_in_new_territory = motif_category_present_in_territory?(motif_categories_territory.motif_category_id, new_territory)
+
+      next unless present_in_new_territory
+
+      if present_in_old_territory
+        puts "Ajout de la catégorie de motifs #{motif_categories_territory.motif_category.name} au nouveau territoire"
+        MotifCategoriesTerritory.create!(territory: new_territory, motif_category: motif_categories_territory.motif_category)
+      else
+        puts "Déplacement de la catégorie de motifs #{motif_categories_territory.motif_category.name} dans le nouveau territoire"
+        # Cette table n'ayant pas d'id, on est obligés de passer par un update_all plutôt qu'un simple update sur le record
```

Comment:
Je n'étais pas au courant de ces limitations sur les tables sans ID, on en apprend tous les jours !

J'ai tenté [d'ajouter une primary key](https://github.com/betagouv/rdv-service-public/compare/frf/add-primary-keys-to-join-tables?expand=1) pour voir si un `update` fonctionnerait mais en fait non. :cry: